### PR TITLE
Add documentation for Helm deployment

### PR DESCRIPTION
**Pull request by sdake**
This adds documentation along with related configuration options for
deploying Istio with the Helm charts.

### CODE REVIEW

**Comment by sdake** [comment]
@sbezverk would you mind reviewing.  There are a couple spots where the documentation has NEEDS REVIEW which could use a one liner description.

File: _docs/setup/kubernetes/helm.md [hunk 1]
```diff
@@ -0,0 +1,76 @@
+---
+title: Alpha Quality Helm Chart
```
Comment (linsun) [inline _docs/setup/kubernetes/helm.md]: we don't normally put Alpha in doc... the status should be tracked in feature stage doc to be consistent

File: _docs/setup/kubernetes/helm.md [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+1. Initialize Helm:
+  ```bash
+  helm init --service-account tiller
+  ```
+
+1. Create the Helm chart:
+  ```bash
+  helm install install/kuberentes/istio --name istio
+  ```
```
Comment (linsun) [inline _docs/setup/kubernetes/helm.md]: typo on kubernetes?

File: _docs/setup/kubernetes/helm.md [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+{% include home.html %}
+
+Quick Start instructions for the setup and configuration of Istio using the Helm package manager.
+
+
+## Prerequisites
+
+The following instructions require you have access to Helm **2.7.2 or newer** in your Kubernetes environment or 
+alternately the ability to modify RBAC rules required to install Helm.
```
Comment (linsun) [inline _docs/setup/kubernetes/helm.md]: Let us refer to k8s version in the setup doc too.

File: _docs/setup/kubernetes/helm.md [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+| istio.deploy_base_config    | true/false         | true                       | NEEDS REVIEW                                 |
+| istio.initializer_enabled   | true/false         | true                       | Specifies whether to use the Initializer     |
+| istio.mixer_enabled         | true/false         | true                       | Specifies whether mixer is enabled/disabled  |
+| istio.pilot_enabled         | true/false	   | true                       | Specifies whether pilot is enabled/disabled  |
+| istio.ca_hub:               | registry+namespace | release registry/namespace | Specifies the HUB for the ca image           |
+| istio.ca_tag:               | image tag          | release unique hash        | Specifies the TAG for the ca image           |
+| istio.ca_demo               | true/false         | false                      | NEEDS REVIEW                                 |
+| istio.ingress.use_nodeport  | true/false         | false                      | Specifies whether to use nodeport or LB      |
+| istio.ingress.nodeport_port | 1024 to 65535      | 32000                      | If nodeport is used, specifies its port      |
+
```
Comment (linsun) [inline _docs/setup/kubernetes/helm.md]: could you update this?  This was changed yesterday to simplify things.

**Comment by sdake** [comment]
@linsun Thanks for the review.  I'm sort of consumed today and tomorrow.  Will produce the requested changes Monday.  By feature tracking, do you mean this document? https://docs.google.com/spreadsheets/d/1mUIrX82Z8mRuhNWyl5D_sd_Evg3KzeNotzRorPzQMko/edit?ts=5a03f88d#gid=351491887

Cheers
-steve

**Comment by ldemailly** [comment]
I believe lin meant https://istio.io/about/feature-stages.html

**Comment by sdake** [comment]
@ldemailly thanks - will address that in the next rev.

**Comment by ldemailly** [comment]
that page itself being quite sensitive - I would just focus on getting working helm instructions and not so much about alpha status or updating the feature stage - that can be later I think

**Comment by linsun** [comment]
Thank you @sdake and @ldemailly !  The plan sounds good.

File: _docs/setup/kubernetes/helm.md (hunk 1 above)
Comment (sdake) [reply _docs/setup/kubernetes/helm.md]: fixed.

File: _docs/setup/kubernetes/helm.md (hunk 2 above)
Comment (sdake) [reply _docs/setup/kubernetes/helm.md]: fixed.

File: _docs/setup/kubernetes/helm.md (hunk 3 above)
Comment (sdake) [reply _docs/setup/kubernetes/helm.md]: fixed.

File: _docs/setup/kubernetes/helm.md (hunk 4 above)
Comment (sdake) [reply _docs/setup/kubernetes/helm.md]: updated.  Agree much simpler.

**Comment by sdake** [comment]
This patch should be good to go.  I am aware of https://github.com/istio/istio/pull/2734, however, I had committed to @costinm to have these documentation changes in around the release of 0.4.0 (~3 weeks ago).  With all of the refactoring related to Helm, it was difficult to find a good point at which to merge these changes.  I think the above PR will take some time to sort out and will commit to revising this documentation when https://github.com/istio/istio/pull/2734 merges.

Cheers
-steve

**Comment by sdake** [comment]
@linsun would you mind merging this?

Cheers
-steve

File: _docs/setup/kubernetes/helm.md [hunk 5]
```diff
@@ -0,0 +1,75 @@
+---
+title: Istio Helm Chart Instructions
+overview: Instructions for the setup and configuration of Istio using the Helm package manager.
+
+order: 10
```
Comment (ldemailly) [inline _docs/setup/kubernetes/helm.md]: order should not be the same as quickstart

can you attach a preview of the toc/nav once fixed

File: _docs/setup/kubernetes/helm.md [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+| global.initializer.enabled   | true/false         | true                       | Specifies whether to use the Initializer       |
+| global.proxy.hub             | registry+namespace | release registry/namespace | Specifies the HUB for the proxy image          |
+| global.proxy.tag             | image tag          | release unique hash        | Specifies the TAG for the proxy image          |
+| global.proxy.debug           | true/false         | false                      | Specifies whether proxy is run in debug mode   |
+| global.pilot.hub             | registry+namespace | release registry/namespace | Specifies the HUB for the pilot image          |
+| global.pilot.tag             | image tag          | release unique hash        | Specifies the TAG for the pilot image          |
+| global.pilot.enabled         | true/false         | true                       | Specifies whether pilot is enabled/disabled     |
+| istio.ca.hub                 | registry+namespace | release registry/namespace | Specifies the HUB for the ca image             |
+| istio.ca.tag                 | image tag          | release unique hash        | Specifies the TAG for the ca image             |
```
Comment (linsun) [inline _docs/setup/kubernetes/helm.md]: do we still have these two helm vars?   think they are renamed to global.security.hub and tag.

**Comment by sdake** [comment]
@linsun @ldemailly thanks for the quick reviews.  I'll take a look today and push an update on security (if that has changed) and attach a nav page this evening.

Cheers
-steve

**Comment by linsun** [comment]
@ldemailly is there a place we can download a daily build?  would be nice to validate the packaging is good in a pre-0.5 build.

**Comment by ldemailly** [comment]
@linsun yes http://gcsweb.istio.io/gcs/istio-prerelease/daily-build/

File: _docs/setup/kubernetes/helm.md (hunk 5 above)
Comment (sdake) [reply _docs/setup/kubernetes/helm.md]: good catch!  Taught me something new.

File: _docs/setup/kubernetes/helm.md (hunk 6 above)
Comment (sdake) [reply _docs/setup/kubernetes/helm.md]: Good catch.

**Comment by sdake** [comment]
Navigation page attached:
screen shot 2018-01-23 at 6 13 01 pm

**Comment by sdake** [comment]
I noticed one small error after testing the instructions after rendering them (thanks @ldemailly for the hint :).  I'll push an additional patch to fix that problem @6pm PST today.  Please don't merge until a 1 liner update comes.

Cheers
-steve

File: _docs/setup/kubernetes/helm.md (hunk 2 above)
Comment (ldemailly) [reply _docs/setup/kubernetes/helm.md]: nope, still kuberenetes

File: _docs/setup/kubernetes/helm.md [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+  ```
+
+1. Initialize Helm:
+  ```bash
+  helm init --service-account tiller
+  ```
+
+1. Create the Helm chart:
+  ```bash
+  helm install install/kuberenetes/helm/istio --name istio
```
Comment (ldemailly) [inline _docs/setup/kubernetes/helm.md]: ping ^ kuberenetes
can you try the command line too / does it work :-)

Comment (sdake) [reply _docs/setup/kubernetes/helm.md]: A sign I need to go to bed :-) - I have run this workflow about 10-15 times in the last 2 days, however, I was manually typing and using completion.  I'll make the typo change, render the docs, and C&P the docs into a log - in the morning :)  Thanks for the patience.

Comment (ldemailly) [reply _docs/setup/kubernetes/helm.md]: you can use the "copy" button from the rendered version and paste in terminal to be sure it's the right command :-) 

thx !

Comment (sdake) [reply _docs/setup/kubernetes/helm.md]: yup - my 10-15 runs were for training purposes - so a CP isn't ideal in that model.

**Comment by sdake** [comment]
Looks like istio/istio#2734 merged sooner than I expected.  I will need to change the table in this document considerably, so will need a bit more time on the docs.  Should be able to wrap them up today however.

Cheers
-steve

**Comment by sdake** [comment]
This review is blocked on sorting out the Helm charts for the 0.5.0 release.  I'll ping reviewers when it has been updated to match the new canonical version.

**Comment by sdake** [comment]
@linsun My plan here is to merge these as is (since these instructions render properly and work today) and follow up with an update to the table as a separate patch.  Could you merge, and I'll start working on the updated table contents as well as integrating the README.md in the istio repository related to Helm?

Cheers
-steve

**Pull request merged**